src/java/picard/util/ClippingUtility.java

```diff
@@ -201,28 +204,26 @@ public static AdapterPair adapterTrimIlluminaPairedReads(final SAMRecord read1,
 
     /**
      * When an adapter is matched in only one end of a pair, we check it again with
-     * stricter thresholds.  If it still matches, then we trim both ends of the read
-     * at the same location.
-      */
+     * stricter thresholds.  If it still matches, then we trim the same number of bases
+     * at both ends of the read.
+     */
     private static boolean attemptOneSidedMatch(final SAMRecord read1,
                                              final SAMRecord read2,
-                                             final int index1,
-                                             final int index2,
+                                             final int numBasesClipped1,
+                                             final int numBasesClipped2,
                                              final int stricterMinMatchBases) {
 
         // Save all the data about the read where we found the adapter match
-        final int matchedIndex = index1 == NO_MATCH ? index2 : index1;
-        final SAMRecord matchedRead = index1 == NO_MATCH ? read2 : read1;
-
+        final int numBasesClipped = numBasesClipped1 == 0 ? numBasesClipped2 : numBasesClipped1;
 
         // If it still matches with a stricter minimum matched bases, then
         // clip both reads
-        if (matchedRead.getReadLength() - matchedIndex >= stricterMinMatchBases) {
-            if (read1.getReadBases().length > matchedIndex) {
-                read1.setAttribute(ReservedTagConstants.XT, matchedIndex + 1);
+        if (stricterMinMatchBases <= numBasesClipped) {
+            if (read1.getReadLength() > numBasesClipped) {
+                read1.setAttribute(ReservedTagConstants.XT, read1.getReadLength() - numBasesClipped + 1);
```

Comment:
I think this is problematic too.  The addition of logic to ensure that each read is long enough that the clip point is within the read is a good idea.  But this changes the clip point to be "number of clipped bases from the end of the read", vs. being at the same position/offset in each read, and I think that's wrong.  We leverage the knowledge that in Illumina sequencing we should see the adapter sequence after the same number of read bases - which when reads are of fixed length, will be the same distance from the end.  But when the reads are of different lengths, it's the distance from the start that is important, not the end.